R/derive_param_tte.R

```diff
@@ -752,7 +752,8 @@ extend_source_datasets <- function(source_datasets,
         full_join(
           mutate(by_groups, temp_dummy = 1),
           mutate(source_datasets[[i]], temp_dummy = 1),
-          by = "temp_dummy"
+          by = "temp_dummy",
+          relationship = "many-to-many"
```

Comment:
Do we have a "history" or issue this particular modification fixes? I'm sure it was done with good reason but based on #1880 only the other changes seemed necessary

Reply:
This is necessary to fix
```
Detected an unexpected many-to-many relationship between `x` and `y`.
i Row 1 of `x` matches multiple rows in `y`.
i Row 1 of `y` matches multiple rows in `x`.
i If a many-to-many relationship is expected, set `relationship = "many-to-many"` to silence this warning.
Backtrace:
  1. testthat::expect_error(...)
       at test-derive_param_tte.R:638:2
````